Src/Plugins/Security/SEC101_036.MySqlConnectionStringValidator.cs

```diff
@@ -54,49 +51,38 @@ public static IEnumerable<ValidationResult> IsValidStatic(Dictionary<string, Fle
         protected override IEnumerable<ValidationResult> IsValidStaticHelper(Dictionary<string, FlexMatch> groups)
         {
             if (!groups.TryGetNonEmptyValue("id", out FlexMatch id) ||
+                !groups.TryGetNonEmptyValue("host", out FlexMatch host) ||
                 !groups.TryGetNonEmptyValue("secret", out FlexMatch secret))
             {
                 return ValidationResult.CreateNoMatch();
             }
 
-            // Our matches can sometimes fail to find a host (due to it being constructed in code)
-            // However, the credentials can still be valid, so we should return "unknown".
-            // Grab the empty host here and then short circuit in dynamic validation.
-
-            var fingerprint = new Fingerprint()
-            {
-                Id = id.Value,
-                Secret = secret.Value,
-            };
-            var validationResult = new ValidationResult();
-
-            if (!groups.TryGetNonEmptyValue("host", out FlexMatch host))
-            {
-                validationResult.Fingerprint = fingerprint;
-                validationResult.ValidationState = ValidationState.Unknown;
-                return new[] { validationResult };
-            }
-
-            FlexMatch unused = null;
-            string port = ParseExpression(RegexEngine, groups["0"], PortRegex, ref unused);
-            string database = ParseExpression(RegexEngine, groups["0"], DatabaseRegex, ref unused);
+            groups.TryGetNonEmptyValue("port", out FlexMatch port);
+            groups.TryGetNonEmptyValue("resource", out FlexMatch resource);
 
             string hostValue = FilteringHelpers.StandardizeLocalhostName(host.Value);
-
-            if (HostsToExclude.Contains(hostValue) ||
-                hostValue.IndexOf("postgres", StringComparison.OrdinalIgnoreCase) != -1)
+            if (hostValue.IndexOf("postgres", StringComparison.OrdinalIgnoreCase) != -1 ||
+                HostsToExclude.Any(hostToExclude => hostValue.IndexOf(hostToExclude, StringComparison.OrdinalIgnoreCase) != -1))
```

Comment:
>HostsToExclude.Any(hostToExclude => hostValue.IndexOf(hostToExclude, StringComparison.OrdinalIgnoreCase) != -1) [](http://example.com/codeflow?start=16&length=111)

we weren't filtering correctly in our previous validations.

Reply:
oh, because these are components of the host rather than the actual hosts. did you add tests for this?

Reply:
we have one dropped with this change, since we will filter correctly